pkg/odo/cli/init/params/interface.go

```diff
@@ -0,0 +1,14 @@
+package params
+
+const FLAG_NAME = "name"
+const FLAG_DEVFILE = "devfile"
+const FLAG_DEVFILE_REGISTRY = "devfile-registry"
+const FLAG_STARTER = "starter"
+const FLAG_DEVFILE_PATH = "devfile-path"
+
+type ParamsBuilder interface {
+	// IsAdequate returns true if the implementation is able to build parameters, given the arguments passed to the comamnd
+	IsAdequate(flags map[string]string) bool
+	// ParamsBuild returns parameters for init
+	ParamsBuild() (InitParams, error)
```

Comment:
In this case, IsAdequate means if the called implementation will be adequate to be called, given the flags passed to the CLI. Insead of having some external code that decides which backend to call, the code asks to all backends (in some specific order) which one wants to do the work, and the first that replies Yes does it.

The second method "Builds the parameters". It could be BuildParams

Reply:
Ack for `IsAdequate`.

Renaming to `BuildParams` makes more sense. 👍🏾